include/swift/AST/DiagnosticsSema.def

```diff
@@ -780,6 +780,9 @@ ERROR(reserved_member_name,none,
       "type member must not be named %0, since it would conflict with the"
       " 'foo.%1' expression", (DeclName, StringRef))
 
+ERROR(invalid_redecl_by_optionality,none,
```

Comment:
Let's use a note for this so we can reduce the verbosity involved in the sequence of diagnostics here. Just a suggestion, but how about

```
// error: invalid redeclaration of 'optionalInOut(x:)'
// note: parameter with %select{implicitly unwrapped|}0 optional type is not different from original parameter with %select{implicitly unwrapped|}1 optional type 
```

Then we can make sure to highlight the parameter in question.

Reply:
Hey @CodaFi, got a little stuck here. Since, there is no longer a function that can be used to determine if it's IUO, and IUO now also falls under `Optional<T>`, do you have any guidance/suggestion as to which way to use to determine whether it's IUO? Also, do correct me if I am wrong in my assumption

Reply:
IUOs are canonically OptionalTypes in the type AST, but you have the `OptionalTypeKind` at your disposal. That can tell you which kind of optional it is.

Reply:
@CodaFi That makes sense. However, I found out that `getAnyOptionalObjectType(OptionalTypeKind kind)` is a function that is currently replaced by `getOptionalObjectType()`, which only returns if it's an `Optional<T>`. I checked if `current` using `isImplicitlyUnwrappedOptional()`, which returns false, which is understandable because when I call `dump` on `current` ValueDecl, it returns the interface type as `Int?`. Do you suggest adding a function (if possible given that ImplicitlyUnwrappedOptional<T> is no longer part of the Type system?

Reply:
Please ignore my last comment, I found a method in swift namespace: `swift::getParameterAt`, from which I was able to get boolean if it's IUO